taprpc/priceoraclerpc/price_oracle.proto

```diff
@@ -121,9 +121,14 @@ message QueryAssetRatesRequest {
     // unit
     AssetSpecifier payment_asset = 4;
 
+    // payment_asset_max_amount is the maximum amount of the payment asset that
+    // could be involved in the transaction. This field is optional. If set to
+    // zero, it is considered unset.
+    uint64 payment_asset_max_amount = 5;
```

Comment:
Do we need this field for an actual use case today? I think we should just remove it to make things easier to understand... See my other comments as well.

Reply:
`MaxInAsset` sometimes relates to BTC and sometimes to a tap asset.

The two different fields in the price oracle relate to the two different assets that we're tying to find an exchange rate for. Instead of a single field that is sometimes the upper bound on the tap asset and sometimes an upper bound on the BTC side, I've got two different fields.

Reply:
Okay, I guess I see where my confusion is coming from. It's because we still do the "flip" of the transaction type for the oracle (which I eliminated in my WIP refactor PR).

What I mean is:
 - On the wallet end user side, a `SellOrder` goes into the oracle as an `ask` (transaction type `sale` with `payment_assset_max_amount` set)
 - On the edge node, that same `SellOrder` goes into the oracle as a `bid` (transaction type `purchase` with `payment_max_amount` set)

And then for a `BuyOrder` it would be kind of the other way around.
Can we please add all of this to the RFQ documentation?

And perhaps update the comments in this proto file to make it more clear to the developer of an oracle when what field is set? For example that never both `_max_amount` fields are set at the same time?